libs/langchain/langchain/vectorstores/semadb.py

```diff
@@ -0,0 +1,267 @@
+from typing import Any, Iterable, List, Optional, Tuple
+from uuid import uuid4
+
+import numpy as np
+import requests
+
+from langchain.schema.document import Document
+from langchain.schema.embeddings import Embeddings
+from langchain.vectorstores import VectorStore
+from langchain.vectorstores.utils import DistanceStrategy
+
+
+class SemaDB(VectorStore):
+    """`SemaDB` vector store.
+
+    This vector store is a wrapper around the SemaDB database.
+
+    Example:
+        .. code-block:: python
+
+            from langchain.vectorstores import SemaDB
+
+            db = SemaDB('mycollection', 768, KEY, embeddings, DistanceStrategy.COSINE)
+
+    """
+
+    HOST = "semadb.p.rapidapi.com"
+    BASE_URL = "https://" + HOST
+
+    def __init__(
+        self,
+        collection_name: str,
+        vector_size: int,
+        api_key: str,
```

Comment:
most integrations make it possible to also specify api keys via environment vars, thoughts on doing that here?
```python
from langchain.utils import get_from_env

class SemaDB(VectorStore):
    def __init__(
        ...
        *,
        api_key: Optional[str] = None,
        distance_strategy: DistanceStrategy = DistanceStrategy.EUCLIDEAN_DISTANCE,
    ) -> None:
        
        self.api_key = api_key or get_from_env("api_key", "SEMA_DB_API_KEY")
```

Reply:
Sure, that makes sense. I'll update the tutorial notebook too.